contracts/YearnCrvTricrypto.sol

```diff
@@ -0,0 +1,120 @@
+  uint256 public constant virtualPriceTol = scale - 510; // 51% drop, since 1000-510=490, and multiplying by 0.49 = 51% drop
+
+  /// @dev Consider trigger toggled if Curve internal balances are lower than true balances by this percentage
+  uint256 public constant balanceTol = scale - 500; // 50% drop
+
+  // --- Trigger Data ---
+  /// @notice Yearn vault this trigger is for
+  IYVaultV2 public immutable vault;
+
+  /// @notice Curve tricrypto pool used as a strategy by `vault`
+  ICrvTricrypto public immutable curve;
+
+  /// @notice Last read pricePerShare
+  uint256 public lastPricePerShare;
+
+  /// @notice Last read curve virtual price
+  uint256 public lastVirtualPrice;
+
+  // --- Constructor ---
+
+  /**
+   * @param _vault Address of the Yearn V2 vault this trigger should protect
+   * @param _curve Address of the Curve Tricrypto pool uses by the above Yearn vault
+   * @dev For definitions of other constructor parameters, see ITrigger.sol
+   */
+  constructor(
+    string memory _name,
+    string memory _symbol,
+    string memory _description,
+    uint256[] memory _platformIds,
+    address _recipient,
+    address _vault,
+    address _curve
+  ) ITrigger(_name, _symbol, _description, _platformIds, _recipient) {
+    // Set vault
+    vault = IYVaultV2(_vault);
+    curve = ICrvTricrypto(_curve);
+
+    // Save current values (immutables can't be read at construction, so we don't use `vault` or `curve` directly)
+    lastPricePerShare = IYVaultV2(_vault).pricePerShare();
+    lastVirtualPrice = ICrvTricrypto(_curve).get_virtual_price();
+  }
+
+  // --- Trigger condition ---
+
+  /**
+   * @dev Checks the yVault pricePerShare
+   */
+  function checkTriggerCondition() internal override returns (bool) {
+    // Read this blocks share price and virtual price
+    uint256 _currentPricePerShare = vault.pricePerShare();
+    uint256 _currentVirtualPrice = curve.get_virtual_price();
+
+    // Check trigger conditions. We could check one at a time and return as soon as one is true, but it is convenient
+    // to have the data that caused the trigger saved into the state, so we don't do that
+    bool _statusVault = _currentPricePerShare < ((lastPricePerShare * vaultTol) / scale);
+    bool _statusVirtualPrice = _currentVirtualPrice < ((lastVirtualPrice * virtualPriceTol) / scale);
+    bool _statusBalances = checkCurveBalances();
+
+    // Save the new data
+    lastPricePerShare = _currentPricePerShare;
+    lastVirtualPrice = _currentVirtualPrice;
+
+    // Return status
+    return _statusVault || _statusVirtualPrice || _statusBalances;
+  }
+
+  /**
+   * @dev Checks if the Curve internal balances are significantly lower than the true balances
+   * @return True if balances are out of tolerance and trigger should be toggled
+   */
+  function checkCurveBalances() internal view returns (bool) {
+    if (usdt.balanceOf(address(curve)) < ((curve.balances(usdtIndex) * virtualPriceTol) / scale)) return true;
```

Comment:
Why if statements instead of:

```solidity
return 
       (usdt.balanceOf(address(curve)) < ((curve.balances(usdtIndex) * virtualPriceTol) / scale)) || 
       (wbtc.balanceOf(address(curve)) < ((curve.balances(wbtcIndex) * virtualPriceTol) / scale)) ||
       (weth.balanceOf(address(curve)) < ((curve.balances(wethIndex) * virtualPriceTol) / scale));
```

Reply:
No reason really, just defaulted to it 🤷

Your way probably saves a tiny amount of gas too, so switched to that in cfaf764e6b88d6594bd3228f7c91cdde0f74786c